internal/storage/field_value.go

```diff
@@ -1029,17 +1031,43 @@ func (ifv *FloatVectorFieldValue) Size() int64 {
 	return int64(len(ifv.Value) * 8)
 }
 
-func NewScalarFieldValueFromGenericValue(dtype schemapb.DataType, gVal *planpb.GenericValue) ScalarFieldValue {
+func NewScalarFieldValueFromGenericValue(dtype schemapb.DataType, gVal *planpb.GenericValue) (ScalarFieldValue, error) {
 	switch dtype {
+	case schemapb.DataType_Int8:
+		i64Val := gVal.Val.(*planpb.GenericValue_Int64Val)
+		if i64Val.Int64Val > math.MaxInt8 || i64Val.Int64Val < math.MinInt8 {
+			return nil, merr.WrapErrParameterInvalidRange(math.MinInt8, math.MaxInt8, i64Val.Int64Val, "expr value out of bound")
+		}
+		return NewInt8FieldValue(int8(i64Val.Int64Val)), nil
+
+	case schemapb.DataType_Int16:
+		i64Val := gVal.Val.(*planpb.GenericValue_Int64Val)
+		if i64Val.Int64Val > math.MaxInt16 || i64Val.Int64Val < math.MinInt16 {
+			return nil, merr.WrapErrParameterInvalidRange(math.MinInt16, math.MaxInt16, i64Val.Int64Val, "expr value out of bound")
+		}
+		return NewInt16FieldValue(int16(i64Val.Int64Val)), nil
+
+	case schemapb.DataType_Int32:
+		i64Val := gVal.Val.(*planpb.GenericValue_Int64Val)
+		if i64Val.Int64Val > math.MaxInt32 || i64Val.Int64Val < math.MinInt32 {
+			return nil, merr.WrapErrParameterInvalidRange(math.MinInt32, math.MaxInt32, i64Val.Int64Val, "expr value out of bound")
+		}
+		return NewInt32FieldValue(int32(i64Val.Int64Val)), nil
 	case schemapb.DataType_Int64:
```

Comment:
1. handle lower/upper bound for int8/16/32
2. separate varchar/string
3. separate double/float